javascript/ql/test/query-tests/Security/CWE-754/UnvalidatedDynamicMethodCallGood3.js

```diff
@@ -0,0 +1,20 @@
+var express = require('express');
+var app = express();
+
+var actions = new Map();
+actions.put("play", function play(data) {
+    // ...
+});
+actions.put("pause", function pause(data) {
+    // ...
+});
+
+app.get('/perform/:action/:payload', function(req, res) {
+    let action = actions.get(req.params.action);
+    // GOOD: `action` is either the `play` or the `pause` function from above
+    if (typeof action === 'function') {
```

Comment:
This should not be flagged because `action` cannot be `proto` right?

Reply:
That sounds right, a call to `get` on a map shouldn't be able to return something from the prototype.   

I've seen a bunch of `get` implementations that are implemented like `function get(x) {return obj[x];}`, but if that's the case, then the query should pick up on the dynamic property read inside the `get` function.

Reply:
That sounds right, a call to `get` on a map shouldn't be able to return something from the prototype.

I've seen a bunch of get implementations that are implemented like `function get(x) {return obj[x];}`, but if that's the case, then the query should pick up on the dynamic property read inside the `get` function.